test/logging/error.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logging
+
+import (
+	"fmt"
+
+	"github.com/davecgh/go-spew/spew"
+)
+
+// StructuredError is an error which can hold arbitrary key-value arguments.
+//
+// Experimental and likely to be removed.
+type StructuredError interface {
```

Comment:
error is an interface, I've had enough headache with structs instead of interfaces (like why I had to create test.T and test.TLegacy and change all the library functions that were using testing.T because it's a struct...)

Reply:
So please resolve if you can live with it.